src/pages/api/og/community/[chain]/[contractAddress]/fcframe.tsx

```diff
@@ -0,0 +1,796 @@
+          </div>
+        ),
+        {
+          width: WIDTH_OPENGRAPH_IMAGE,
+          height: HEIGHT_OPENGRAPH_IMAGE,
+          fonts: [
+            {
+              name: 'ABCDiatype-Regular',
+              data: ABCDiatypeRegularFontData,
+              weight: 400,
+            },
+            {
+              name: 'ABCDiatype-Bold',
+              data: ABCDiatypeBoldFontData,
+              weight: 700,
+            },
+            {
+              name: 'GT Alpina',
+              data: alpinaLightFontData,
+              style: 'normal',
+              weight: 500,
+            },
+          ],
+        },
+      );
+    }
+  } catch (e) {
+    console.log('error: ', e);
+    return fallbackImageResponse;
+  }
+};
+
+export default handler;
+
+type ImagePosition = { left: number; top: number };
+
+function calcRandomPosition({
+  // the image we want to place
+  elementSize,
+  // the size of the entire container
+  containerSize,
+  // the area where the text goes
+  textAreaBoundingBox,
+  // the positions of existing images
+  existingPositions,
+}): ImagePosition | null {
+  let position;
+  let isOverlapping;
+
+  // check if the generated position overlaps with the text area or other photos
+  function checkOverlap(pos) {
+    // safety padding
+    const buffer = 20;
+    if (
+      pos.left < textAreaBoundingBox.right + buffer &&
+      pos.left + elementSize.width > textAreaBoundingBox.left - buffer &&
+      pos.top < textAreaBoundingBox.bottom + buffer &&
+      pos.top + elementSize.height > textAreaBoundingBox.top - buffer
+    ) {
+      return true;
+    }
+
+    // check against other images
+    for (const existingPosition of existingPositions) {
+      if (
+        pos.left < existingPosition.left + elementSize.width + buffer &&
+        pos.left + elementSize.width > existingPosition.left - buffer &&
+        pos.top < existingPosition.top + elementSize.height + buffer &&
+        pos.top + elementSize.height > existingPosition.top - buffer
+      ) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  // try to find a non-overlapping position
+  const maxAttempts = 1000;
+  for (let i = 0; i < maxAttempts; i++) {
+    position = {
+      left: Math.floor(Math.random() * (containerSize.width - elementSize.width)),
+      top: Math.floor(Math.random() * (containerSize.height - elementSize.height)),
+    };
+
+    isOverlapping = checkOverlap(position);
+
+    if (!isOverlapping) {
+      break;
+    }
+  }
+
+  return isOverlapping ? null : position;
+}
+
+// TODOs:
+// - art blocks / prohibition support *
+// - correctly estimate rendered text dimensions ***
+// might just make the community title be max-1-line-tall for now (i know we wanted to allow for 2 lines)
+// - different aspect ratios (tall vs. flat)
+// letting images get cropped off the sides (right now they’re forced to be contained within the frame
+// - post request to flip through different tokens in that collection
```

Comment:
so i think these are all done so we can remove this block, with the exception of L757- art blocks / prohibition support. those have a different URL scheme under `/community/artblocks/...` as kaito recently updated in this PR: https://github.com/gallery-so/gallery/pull/2333

let's handle AB/prohibition in a follow-up! not as high priority as generalizing the aspect ratio to other frames though